app/models/watchlist_instance.rb

```diff
@@ -515,10 +515,16 @@ def self.add_new_instance_for_cud_grace_day_usage(course, condition_id,
 
   def self.add_new_instance_for_cud_extension_requests(course, category_blocklist,
                                                        condition_id, cud, extension_count)
-    asmts = Assessment.where.not(category_name: category_blocklist).pluck(:id)
+    asmts = Assessment.where.not(category_name: category_blocklist)
     violation_info = {}
     num_of_extensions = 0
 
+    date = DateTime.current
+    asmts = asmts.reject do |asmt|
+      asmt.due_at > date
+    end
+    asmts = asmts.pluck(:id)
```

Comment:
nit: looks like this logic is repeated a few times in this file, might want to just create a private function or something

Reply:
Okay, fixed.